### ISSUE TITLE
2D Convolution operations

### ISSUE THREAD

**Issue by dvigneshwer** [dvigneshwer/deeprust#1]
![image](https://cloud.githubusercontent.com/assets/5075497/20252210/50d4b74a-aa46-11e6-8d2c-b9cd37b01401.png)

Convolutional Neural networks(CNNs) consists of one or more convolutional layers
with some sub-sampling function followed by one or two fully connected layers. CNNs
are designed to avoid the situation of handling billions of parameters as in case of fully
connected networks with the same number of neurons. CNNs work well with images
due to the local connections and tied weights followed by some form of pooling which
results in translation invariant features.

![image](https://cloud.githubusercontent.com/assets/5075497/20252270/a4791a76-aa46-11e6-9639-8404d4364e6d.png)

The number of convolutional layers determine the level of features, the first layer can
extract features like edges and lines and with the increase in the number of layer these
features combine to give high-level features. As discussed earlier all the neurons in
this layer are not connected to each other instead they are connected to a local region.
The spatial extent of this connectivity is a hyperparameter called the receptive field
of the neuron. The connections are local in space (width and height) but always full
along the depth i.e. if the dimensions of the input are n × n × d and dimensions of
the kernel are m × m × r then m < n but r = d . The other hyperparameters for
this layer are stride and zero padding. Stride determines the amount of overlapping
of the receptive fields i.e. number of units the kernel shifts while convolution and zero
padding is the amount by which input is padded with zeros spatially on the border,
this is an important factor to determine the volume of the output.

**Comment by siddas27** [comment]
What's the task exactly?
